packages/babel-helpers/src/helpers.js

```diff
@@ -447,8 +447,15 @@ helpers.inheritsLoose = () => template.program.ast`
 helpers.wrapNativeSuper = () => template.program.ast`
   var _gPO = Object.getPrototypeOf || function _gPO(o) { return o.__proto__ };
   var _sPO = Object.setPrototypeOf || function _sPO(o, p) { o.__proto__ = p; return o };
-  var _construct = (typeof Reflect === "object" && Reflect.construct) ||
-    function _construct(Parent, args, Class) {
+  
+  // Here we don't use (Reflect.construct || function _construct(...) {})
+  // because, when running the exec tests in node 4, there is a buggy
+  // Reflect.construct implementation which ignores the third parameter.
+  // I don't know why, since Reflect shouldn't exists on node 4 (and it
+  // doesn't exist if you try to run the compiled code, it only exists in exec
+  // tests).
+  // @nicolo-ribaudo
+  function _construct(Parent, args, Class) {
```

Comment:
Whoa, can’t do this. `Reflect.construct` cannot be properly polyfilled, but it’s necessary for non-constructable constructors (like `HTMLElement`).

Reply:
I think that it won't work anyway:
```js
> function Wrapper() {
    return Reflect.construct(HTMLElement, [], Wrapper);
  }
  Wrapper.__proto__ = HTMLElement;
  Wrapper.prototype = Object.create(HTMLElement.prototype, { constructor: { value: Wrapper, configurable: true } })

-> HTMLElement {constructor: ƒ}

> new Wrapper()

-> VM195:2 Uncaught TypeError: Illegal constructor
     at new Wrapper (<anonymous>:2:18)
     at <anonymous>:1:1
   Wrapper @ VM195:2
   (anonymous) @ VM199:1
```

Reply:
It's because you can't directly construct an `HTMLElement`, it has to go through the custom elements APIs.

```js
function Wrapper() {
  return Reflect.construct(HTMLElement, [], Wrapper);
}
Wrapper.__proto__ = HTMLElement;
Wrapper.prototype = Object.create(HTMLElement.prototype, { constructor: { value: Wrapper, configurable: true } })

customElements.define('x-w', Wrapper)

// no errors should be raised
w = document.createElement('x-w');
```

As opposed to using a `bind` constructor:

```js
function Binder() {
    var Constructor, a = [null];
    a.push.apply(a, arguments);
    Constructor = HTMLElement.bind.apply(HTMLElement, a);
    return Object.setPrototypeOf(new Constructor, Binder.prototype);
}
Binder.__proto__ = HTMLElement;
Binder.prototype = Object.create(HTMLElement.prototype, { constructor: { value: Binder, configurable: true } })

customElements.define('x-b', Binder)

// TypeError: Illegal constructor
b = document.createElement('x-b');
```

Reply:
Back when this was first implemented, I had actually checked if node 4 had a Reflect.construct (because I know node 4 has buggy classes) and it did not have Reflect at all.

Is core-js maybe providing a buggy one? Or is the one they provide doing approximately the same thing as the helper, and thus would not work just like the helper wouldn't?

Reply:
It's [doing](https://github.com/zloirock/core-js/blob/master/modules/es6.reflect.construct.js) roughly what we are.